iris-mpc-common/src/helpers/batch_sync.rs

```diff
@@ -11,6 +13,49 @@ pub struct BatchSyncState {
     pub batch_id: u64,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct BatchSyncEntries {
+    pub valid_entries: Vec<bool>,
+    pub batch_sha: [u8; 32],
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct BatchSyncEntriesResult {
+    pub my_state: BatchSyncEntries,
+    pub all_states: Vec<BatchSyncEntries>,
+}
+
+impl BatchSyncEntriesResult {
+    pub fn new(my_state: BatchSyncEntries, all_states: Vec<BatchSyncEntries>) -> Self {
+        Self {
+            my_state,
+            all_states,
+        }
+    }
+    pub fn sha_matches(&self) -> bool {
+        self.all_states
+            .iter()
+            .all(|s| s.batch_sha == self.my_state.batch_sha)
+    }
+
+    pub fn valid_entries(&self) -> Vec<bool> {
```

Comment:
- For each party’s vector, iterate over its entries; whenever any party has false at index i, set the accumulator at i to false.
- After folding all states, the accumulator contains true only for indices that were true in every party.